lib/compress/zstd_compress.c

```diff
@@ -1536,6 +1536,12 @@ MEM_STATIC size_t ZSTD_compressSequences(seqStore_t* seqStorePtr,
     int const uncompressibleError = (cSize == ERROR(dstSize_tooSmall)) && (srcSize <= dstCapacity);
     if (ZSTD_isError(cSize) && !uncompressibleError)
         return cSize;
+    /* We check that dictionaries have offset codes available for the first
+     * block. After the first block, the offcode table might not have large
+     * enough codes to represent the offsets in the data.
+     */
+    if (entropy->offcode_repeatMode == FSE_repeat_valid)
+        entropy->offcode_repeatMode = FSE_repeat_check;
```

Comment:
I'm not completely sure to understand what this line does.
What's the meaning of `FSE_repeat_check` ?
It seems to make "repeat" mode selection impossible. Is that correct ?

If yes, what about the first block ? Can it still benefit from dictionary's offset table ?

Reply:
This line is the equivalent of [this line](https://github.com/facebook/zstd/commit/830ef4152a8b7568916ecb600a0aac1a5212c08f#diff-62f99f3a22fe685ec153abb360cca5d5L1136), but only for offcodes. Its called after `ZSTD_compressSequences_internal()` finishes, so the first block can still use the dictionaries table (I just doubled checked that it is used).

Right now, we don't do anything with `check`, it is the same as `none`. The `valid` repeat mode means you can assume the table is valid. `check` means a table exists, but you need to ensure that it has non-zero probabilities for all the symbols you require. `none` means we don't have a table to repeat (first block, or previous block was incompressible).